autoPyTorch/data/utils.py

```diff
@@ -0,0 +1,344 @@
+            )
+        # convert to int so we can directly use
+        dataset_compression["memory_allocation"] = floor(memory_allocation * memory_limit)
+
+    # "methods" must be non-empty sequence
+    if (
+        not isinstance(dataset_compression["methods"], Sequence)
+        or len(dataset_compression["methods"]) <= 0
+    ):
+        raise ValueError(
+            "key 'methods' must be a non-empty list"
+            f"\nmethods = {dataset_compression['methods']}"
+            f"\ndataset_compression = {dataset_compression}"
+        )
+
+    # "methods" must contain known methods
+    if any(
+        method not in cast(Sequence, default_dataset_compression_arg["methods"])  # mypy
+        for method in dataset_compression["methods"]
+    ):
+        raise ValueError(
+            f"key 'methods' can only contain {default_dataset_compression_arg['methods']}"
+            f"\nmethods = {dataset_compression['methods']}"
+            f"\ndataset_compression = {dataset_compression}"
+        )
+
+    return cast(DatasetCompressionSpec, dataset_compression)
+
+
+class _DtypeReductionMapping(Mapping):
+    """
+    Unfortuantly, mappings compare by hash(item) and not the __eq__ operator
+    between the key and the item.
+
+    Hence we wrap the dict in a Mapping class and implement our own __getitem__
+    such that we do use __eq__ between keys and query items.
+
+    >>> np.float32 == dtype('float32') # True, they are considered equal
+    >>>
+    >>> mydict = { np.float32: 'hello' }
+    >>>
+    >>> # Equal by __eq__ but dict operations fail
+    >>> np.dtype('float32') in mydict # False
+    >>> mydict[dtype('float32')]  # KeyError
+
+    This mapping class fixes that supporting the `in` operator as well as `__getitem__`
+
+    >>> reduction_mapping = _DtypeReductionMapping()
+    >>>
+    >>> reduction_mapping[np.dtype('float64')] # np.float32
+    >>> np.dtype('float32') in reduction_mapping # True
+    """
+
+    # Information about dtype support
+    _mapping: Dict[type, type] = {
+        np.float32: np.float32,
+        np.float64: np.float32,
+        np.int32: np.int32,
+        np.int64: np.int32
+    }
+
+    # In spite of the names, np.float96 and np.float128
+    # provide only as much precision as np.longdouble,
+    # that is, 80 bits on most x86 machines and 64 bits
+    # in standard Windows builds.
+    _mapping.update({getattr(np, s): np.float64 for s in ['float96', 'float128'] if hasattr(np, s)})
+
+    @classmethod
+    def __getitem__(cls, item: type) -> type:
+        for k, v in cls._mapping.items():
+            if k == item:
+                return v
+        raise KeyError(item)
+
+    @classmethod
+    def __iter__(cls) -> Iterator[type]:
+        return iter(cls._mapping.keys())
+
+    @classmethod
+    def __len__(cls) -> int:
+        return len(cls._mapping)
+
+
+reduction_mapping = _DtypeReductionMapping()
+supported_precision_reductions = list(reduction_mapping)
+
```

Comment:
```suggestion
FLOAT_DTYPES = tuple(dt for dt in reduction_mapping._mapping.keys() if dt.__name__.startswith("float"))
INT_DTYPES = tuple(dt for dt in reduction_mapping._mapping.keys() if dt.__name__.startswith("int"))
```

Reply:
we are not going to be using these anywhere else. Lets keep it the way you suggested on mattermost.